src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java

```diff
@@ -420,6 +420,9 @@ void saveProgram() throws IOException {
 
     @CpsVmThreadOnly
     public void saveProgram(File f) throws IOException {
+        boolean logging = LOGGER.isLoggable(Level.FINER);
```

Comment:
True -- I couldn't remember if this was one of the methods overridden to do a deceptively expensive lookup.  Java logging can be terrible sometimes. 

I take back my nit after looking more closely at the code path involved here.